llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -643,11 +643,16 @@ static bool isKnownNonZeroFromAssume(const Value *V, const SimplifyQuery &Q) {
     // Warning: This loop can end up being somewhat performance sensitive.
     // We're running this loop for once for each value queried resulting in a
     // runtime of ~O(#assumes * #values).
+    Value *Arg = I->getArgOperand(0);
```

Comment:
Which test covers this change? InstSimplify can fold the zero check in `nonnull_trunc_true` even without the change to `isKnownNonZeroFromAssume`.

Reply:
I'm not sure I understand if I remove this code the tests `nonnull_trunc_true ` and `test4` fails

Reply:
However, InstCombine can simplify both of them after I removed the change to `isKnownNonZeroFromAssume`.
(the original tests check InstSimplify/LVI instead).

Can you please spill `isKnownNonZeroFromAssume` changes into a separate PR? Sorry for the inconvenience.

Reply:
sure I can split this PR to the two commits.